cmake/Platform/Mac/Install_mac.cmake

```diff
@@ -70,31 +64,74 @@ function(ly_install_target_override)
             COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
     )
 
+    set(install_relative_binaries_path "${ly_platform_install_target_RUNTIME_DIR}/${ly_platform_install_target_RUNTIME_SUBDIR}")
+
     if (${is_bundle})
         set_property(TARGET ${ly_platform_install_target_TARGET} PROPERTY RESOURCE ${cached_resources_dir})
+        set(runtime_output_filename "$<TARGET_FILE_NAME:${ly_platform_install_target_TARGET}>.app")
+
+        # For bundles, add a setreg file that contains install relative path to the bundle.
+        # This will be used when loading dynamic libraries outside the bundle.
+        string(CONFIGURE "${installed_binaries_path_template}" configured_setreg_file)
+        file(GENERATE
+            OUTPUT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${ly_platform_install_target_TARGET}BundlePath.setreg
```

Comment:
We are going to have paths would need to take the build permutation into account when installing these files with @amznestebanpapp work in this https://github.com/o3de/o3de/commit/2c066a81daec45277246757ec8be7fd5dfc068e4

Reply:
Do you mean the path to which we're generating the files themselves? The generated files already take "Default"/"Monolithic" into account.

Reply:
not sure why we are generating something in the binary dir at this point, this si apart of the install therefore it should go to an install folder
This bundle path setreg should likely be copied to the output as we copy other registry/target files and then marked to install (note we now have functions so targets can add things to install)

Reply:
Just reread this. This setreg needs to be inside the bundle. It can't be accessed if it's outside the bundle since this setreg's contents are going to be used to determine where the binaries are relative to the install root. It's generated here because this is where we have access to the path where the dynamic lib/exec/bundle is installed to. Otherwise we'd have to duplicate the logic to figure out this path. It's easier if Install_common.cmake is the only place that determines that path.

Reply:
This setreg is only being used to load dylibs outside bundles in Editor/AP. But when we build the project using the SDK, then all those dylibs will be copied to the project build path also I think? If yes, then the project build path is already in the registry and we can just load our dylibs from there. That way, I can get rid of this setreg file altogether.

Reply:
I would still need that setreg file for the O3DE_SDK launcher(main app bundle for SDK) though. Just not for Editor/AP

Reply:
Since the user will have to build the project before running AssetProcessor or Editor to get the dependencies, and the build process copies all the dependencies to the project build folder anyway, I got rid of this setreg for our tools. Instead, the setreg is now only used by the O3DE_SDK.app bundle to locate and launch the ProjectManager app. I moved its generation outside this function.